R/dplyr_methods.R

```diff
@@ -867,36 +867,35 @@ full_join.Seurat <- function (x, y, by = NULL, copy = FALSE, suffix = c(".x", ".
 #' @rdname dplyr-methods
 #' @name slice
 #'
-#' @export
 #' @examples
 #'
 #' `%>%` = magrittr::`%>%`
 #' data("pbmc_small")
 #' pbmc_small %>%  slice(1)
+#'
+#' # Slice group-wise using .by
+#' pbmc_small |> slice(1:2, .by = groups)
 NULL
 
 #' @export
-slice.Seurat <- function (.data, ..., .preserve = FALSE)
+slice.Seurat <- function (.data, ..., .by = NULL, .preserve = FALSE)
 {
-  new_meta = dplyr::slice(.data[[]], ..., .preserve = .preserve)
-
-  # Error if size == 0
-  if(nrow(new_meta) == 0) stop("tidyseurat says: the resulting data container is empty. Seurat does not allow for empty containers.")
-
-  new_obj = subset(.data,   cells = rownames(new_meta ))
-  #new_obj@meta.data = new_meta
-
+  idx <- .data[[]] |>
+    tibble::rowid_to_column(var  = 'row_number___')  |>
+    dplyr::select(-everything(), row_number___, {{ .by }}) |>
```

Comment:
Thanks a lot!

Sorry if I am being pedantic here, but it would be better to do `select` first and `rowid_to_column` later both from a memory and elegance perspective.

Another little feedback,  standard we use is to use 

`@importFrom tibble ...`

and not use 

`tibble:: ...`

With these two changes I think the PR is good to go!

Reply:
No problem, will do